tests/foreman/cli/test_user.py

```diff
@@ -95,7 +95,7 @@ def test_positive_CRUD(self):
             'login': random.choice(valid_usernames_list()),
             'firstname': random.choice(valid_usernames_list()),
             'lastname': random.choice(valid_usernames_list()),
-            'mail': random.choice(valid_emails_list()).replace('"', r'\"').replace('`', r'\`'),
+            'mail': random.choice(valid_emails_list()),
```

Comment:
From what I see, `valid_emails_list` **can** return both `"` and ``` ` ```, and these parameters will be put inside `""` by robottelo, so escaping these two characters before passing to shell does sound like good idea.

At the same time, I agree this escaping should be done on framework level, not on individual test level. But it still should be done, and this PR doesn't address that.

Reply:
it would be useful to see the command formatting before and after the change

Reply:
@mirekdlugosz & @JacobCallahan so the test failed like this:
```
> user_params[key], user[key], 'values for key "{}" do not match'.format(key)
                )
E AssertionError: '!#$%&*+-/=?^\\`{|}~@example.com' != '!#$%&*+-/=?^`{|}~@example.com'
E - !#$%&*+-/=?^\`{|}~@example.com
E ? -
E + !#$%&*+-/=?^`{|}~@example.com
E : values for key "email" do not match
```
The backslash before the backtick is not tolerated, the same thing for double quotes. You see it's wrapped in singlequotes

Reply:
@pondrejk I applied your patch and modified `valid_emails_list` to only return last element (one with backtick). Then I run the test, and it failed:
```
2020-09-01 10:10:37 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=csv user create --auth-source-id="1" --description="RoQXPHfZsbGQTUIoAjCuyagjvJodTjnCXkXIAltHsbJkOLfMuRDCXxtIDYDgjhmXAcvWlZhjFGTSbqq" --firstname="\xc3\xae\xc3\xa3\xc3\x81\xc3\x82\xc3\xaa\xc3\xba\xc3\x91\xc3\x84\xc3\xaf\xc3\x8a\xc3\xa6\xc3\x91\xc3\x9f" --lastname="2683258190944370263347" --login="\xf0\x96\xa2\x97\xec\xb6\xb6\xf0\x93\x8b\x81\xe9\x93\x82\xe3\xb1\xa4\xf0\x97\xb5\xac" --mail="!#$%&*+-/=?^`{|}~@example.com" --password="G2jTsEonsJ"'
2020-09-01 10:10:39 - robottelo.ssh - INFO - <<< stderr
bash: -c: line 0: unexpected EOF while looking for matching ``'
bash: -c: line 1: syntax error: unexpected end of file
```

Look at `--mail` - value is wrapped in double quotes in shell, and bash doesn't like that there is opening backtick, but not closing one. If there was closing one, bash would interpret it as subshell, and we want to pass it on to Satellite.

This PR, in current form, is trading one intermittent failure for another.

I can see couple of possible solutions:

1. Modify robottelo hammer code to escape ``` ` ``` and `"` before running command. This way, test shouldn't fail since input string remains unmodified in test scope. And as mentioned above, individual tests shouldn't need to worry about this.
2. Keep 2 email values in test - one (escaped) to pass to hammer, and one (raw) to run assertions on. It's little ugly, but easier to implement.
3. Modify test so it never uses values with ``` ` ``` and `"` as input for email. We will lose some coverage, but character escaping in Satellite is unlikely to suddenly break, and invalid values are probably better tested on unit level.

Reply:
@mirekdlugosz thanks for this, I went with option 2, I guess it is not such a wide concern to warrant 1